presto-native-execution/presto_cpp/main/TaskManager.h

```diff
@@ -103,6 +103,23 @@ class TaskManager {
       std::optional<protocol::Duration> maxWait,
       std::shared_ptr<http::CallbackRequestHandlerState> state);
 
+  /// Update buffer count for a given task. Returns nullptr, if update was
+  /// successful, else returns a std::unique_ptr to TaskInfo. Note: This method
+  /// is made public for unit testing purpose only.
+  inline std::unique_ptr<protocol::TaskInfo> updateBroadcastOutputBuffers(
```

Comment:
Is it possible to return and test a boolean for this method instead ? Returning nullptr for a successful update vs TaskInfo for a failure seems weird.